### PR TITLE
added support for comments in ServerSentEvent

### PR DESCRIPTION
**Pull request by fizzy33**
A few notes.  

* unit tests had dropLast added because fs2.text.lines sees "\n" as two lines so decode sees it as two events when in fact it is one.  There is an fs2.text conversation happening now over an alternative linesWithEndings that would provide a cleaner path
* note one alternative to dropLast is we always add an extra empty event (maybe cleaner and safer since we could be dropping a non-empty buggy event)
* @rossabaker I will save the cleaner version of decode for now, it just didn't perform as well and I ran out of time for today as discussed the big thing is getting ServerSentEvents with the correct fields
* slight tweak to use an internal LineBuffer performance is on par with StringBuilder albeit a bit more memory consumed again could tweak this with something that accumulates in a StringBuilder and notes if any lines at all have been added (performance is from very informal testing)
* noting in real world I doubt the performance of this will be a gating factor

### CODE REVIEW

File: core/src/main/scala/org/http4s/ServerSentEvent.scala [hunk 1]
```diff
@@ -24,13 +25,15 @@ import org.http4s.util.{Renderable, Writer}
 import scala.util.Try
 
 final case class ServerSentEvent(
-    data: String,
+    data: Option[String] = None,
     eventType: Option[String] = None,
     id: Option[EventId] = None,
-    retry: Option[Long] = None
+    retry: Option[Long] = None,
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/ServerSentEvent.scala:31]: Do you want the `FiniteDuration` before 0.22 ABI gets locked?

**Review by rossabaker — APPROVED** [review]
Looks like progress.  I'll cherry-pick this to 0.22 so it's in that RC today.

File: core/src/main/scala/org/http4s/ServerSentEvent.scala [hunk 2]
```diff
@@ -24,13 +25,15 @@ import org.http4s.util.{Renderable, Writer}
...
+    comments: Option[String] = None,
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/ServerSentEvent.scala:32]: Bikeshed: the plural name is weird to me.  It's just one comment, that might be split across multiple lines.

(`data` is wired into the spec, so I won't be a nerd and try to call that `datum`.)

Comment (fizzy33) [reply core/src/main/scala/org/http4s/ServerSentEvent.scala:32]: "comment" has been pushed

File: core/src/main/scala/org/http4s/ServerSentEvent.scala (hunk 1 above)
Comment (fizzy33) [reply core/src/main/scala/org/http4s/ServerSentEvent.scala:31]: why yes I do thanks for remembering

Comment (fizzy33) [reply core/src/main/scala/org/http4s/ServerSentEvent.scala]: pushed

**Review by rossabaker — APPROVED** [review]

**Comment by rossabaker** [comment]
Picked to 0.22.

**Pull request closed without merging**